pkg/controller/util_test.go

```diff
@@ -1700,6 +1724,15 @@ func createTargetPod(pvc *v1.PersistentVolumeClaim, pvcUID, podAffinityNamespace
 			},
 		},
 	}
+	volumeMode := pvc.Spec.VolumeMode
```

Comment:
This is where you are getting your npe.
So instead of
```go
volumeMode := pvc.Spec.VolumeMode
```
Do
```go
var volumeMode v1.PersistentVolumeMode
if pvc.Spec.VolumeMode != nil {
   volumeMode = *pvc.Spec.VolumeMode
} else {
    volumeMode = v1.PersistentVolumeFilesystem
}

if volumeMode == v1.PersistentVolumeBlock {
...
}
```

And do this for all 4 places where you have this problem, the test will pass.